packages/grafana-runtime/src/services/pluginExtensions/links.test.ts

```diff
@@ -0,0 +1,61 @@
+import { getPluginLink, PluginLinkMissingError } from './links';
+import { setPluginsExtensionRegistry } from './registry';
+
+describe('getPluginLink', () => {
+  describe('when getting a registered extension link', () => {
+    const pluginId = 'grafana-basic-app';
+    const linkId = 'declare-incident';
+
+    beforeAll(() => {
+      setPluginsExtensionRegistry({
+        links: {
+          [`${pluginId}.${linkId}`]: {
+            description: 'Declaring an incident in the app',
+            href: `/a/${pluginId}/declare-incident`,
+          },
+        },
+      });
+    });
+
+    it('should return a href to the plugin', () => {
+      const { link, error } = getPluginLink({
+        id: `${pluginId}.${linkId}`,
+      });
```

Comment:
this does not seem like something Grafana can use directly. But maybe it's only for other plugins to link to another plugin they know is installed? But if they know it's installed they could just link to it, so confused :confused:

Reply:
The things you are requesting will be added in follow up PRs. This first PR will mainly solve the issue of cross-plugin linking. 

It is not that easy to just link to another plugin. The plugin that you want to link from needs to know that the target plugin is installed, enabled and supports the route that you want to link to. Otherwise you might redirect the user to a 404. This functionality will handle all of that for you so you just can conditionally render the link if the route actually exists.